pixeltable/exprs/row_builder.py

```diff
@@ -448,7 +448,14 @@ def create_table_row(self, data_row: DataRow, exc_col_ids: set[int]) -> tuple[di
             else:
                 if col.col_type.is_image_type() and data_row.file_urls[slot_idx] is None:
                     # we have yet to store this image
-                    filepath = str(MediaStore.prepare_media_path(col.tbl.id, col.id, col.tbl.version))
+                    filepath = str(
+                        MediaStore.prepare_media_path(
+                            tbl_id=col.tbl.id, 
+                            col_id=col.id, 
+                            version=col.tbl.version, 
+                            ext=".jpeg",
```

Comment:
It's not necessarily a jpeg; it'll be stored as webp if the image contains transparency data. (See code in flush_img which is called by this function.)

One solution would be to append the extension in `flush_img` instead, once we know the actual filetype. Another could be to set the extension here via

`ext=('webp' if image.has_transparency_data else 'jpeg')`

but that seems less desirable, since it duplicates the logic for determining the image format.

Reply:
To add to Aaron's comments:

DataRow.flush_img() is the correct place for a fix.